src/Datadog.Trace.ClrProfiler.Native/rejit_handler.cpp

```diff
@@ -0,0 +1,164 @@
+#include "rejit_handler.h"
+
+namespace trace {
+
+void RejitHandlerModuleMethod::AddFunctionId(FunctionID functionId) {
+  std::lock_guard<std::mutex> guard(functionsIds_lock);
+  auto moduleHandler = (RejitHandlerModule*)module;
+  auto rejitHandler = (RejitHandler*)moduleHandler->GetHandler();
+  functionsIds.insert(functionId);
+  rejitHandler->_addFunctionToSet(functionId, this);
+}
+bool RejitHandlerModuleMethod::ExistFunctionId(FunctionID functionId) {
+  std::lock_guard<std::mutex> guard(functionsIds_lock);
+  return functionsIds.find(functionId) != functionsIds.end();
+}
+
+RejitHandlerModuleMethod* RejitHandlerModule::GetOrAddMethod(mdMethodDef methodDef) {
+  std::lock_guard<std::mutex> guard(methods_lock);
+
+  if (methods.count(methodDef) > 0) {
+    return methods[methodDef];
+  }
+
+  RejitHandlerModuleMethod* methodHandler = new RejitHandlerModuleMethod(methodDef, this);
+  methods[methodDef] = methodHandler;
+  return methodHandler;
+}
+bool RejitHandlerModule::TryGetMethod(mdMethodDef methodDef,
+                                      RejitHandlerModuleMethod** methodHandler) {
+  std::lock_guard<std::mutex> guard(methods_lock);
+
+  if (methods.count(methodDef) > 0) {
+    *methodHandler = methods[methodDef];
+    return true;
+  }
+  *methodHandler = nullptr;
+  return false;
+}
+
+RejitHandlerModuleMethod* RejitHandler::GetModuleMethodFromFunctionId(
+    FunctionID functionId) {
+  {
```

Comment:
Are those braces needed?

Reply:
yes, those braces limits the effect of the lock `std::lock_guard<std::mutex> guard(methodByFunctionId_lock);`
so the lock is applied inside those braces only.